src/main/java/com/flightstats/hub/metrics/JVMMetricsService.java

```diff
@@ -0,0 +1,118 @@
+package com.flightstats.hub.metrics;
+
+import com.flightstats.hub.app.HubProperties;
+import com.flightstats.hub.app.HubVersion;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
+import com.codahale.metrics.jvm.CachedThreadStatesGaugeSet;
+import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
+import com.codahale.metrics.ScheduledReporter;
+import metrics_influxdb.InfluxdbProtocol;
+import metrics_influxdb.InfluxdbReporter;
+import metrics_influxdb.HttpInfluxdbProtocol;
+import metrics_influxdb.UdpInfluxdbProtocol;
+import com.google.inject.Singleton;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+import java.util.SortedSet;
+import java.util.concurrent.TimeUnit;
+import static com.flightstats.hub.app.HubHost.getLocalName;
+
+@Singleton
+public class JVMMetricsService {
+    private final Logger logger = LoggerFactory.getLogger(JVMMetricsService.class);
+
+    @Inject
+    private MetricRegistry registry;
+
+    @Inject
+    private HubVersion hubVersion;
+
+    private ScheduledReporter influxReporter;
+    private final String protocolScheme = HubProperties.getProperty("influx.protocolScheme", "http");
+    private final String influxHost = HubProperties.getProperty("influx.host", "influxdb");
+    private final int influxPort = HubProperties.getProperty("influx.port", 8086);
+    private final String influxUser = HubProperties.getProperty("influx.dbUser", "");
+    private final String influxPass = HubProperties.getProperty("influx.dbPass", "");
+    private final String influxDatabaseName = HubProperties.getProperty("influx.dbName", "hubmain");
+
+    @Inject
+    JVMMetricsService(MetricRegistry registry, HubVersion hubVersion) {
+        this.hubVersion = hubVersion;
+        this.registry = registry;
+        if (HubProperties.getProperty("metrics.enable", false)) {
+            String env = HubProperties.getProperty("app.environment", "dev");
+            String clusterName = HubProperties.getProperty("cluster.tagName", "single.local");
+            // custom configurable reporter
+            influxReporter = InfluxdbReporter.forRegistry(registry)
```

Comment:
I think you could move the `InfluxdbReporter` creation out into a provider in `HubBindings`. Then you can inject it, which makes mocking in tests a bit easier....that might not be necessary in this class, but it's a good practice to do as little in constructors as possible.

Reply:
makes sense